src/opm/input/eclipse/EclipseState/SimulationConfig/ThresholdPressure.cpp

```diff
@@ -273,4 +280,32 @@ namespace Opm {
                full_arg.m_pressureTable == rst_arg.m_pressureTable;
     }
 
+    void ThresholdPressure::logPressures() {
+        if (!this->active()) {
+            return;
+        }
+        if (!std::any_of(m_pressureTable.begin(), m_pressureTable.end(),
+                         [](const auto& in) {
+                              return in.second.first;
+                         })) {
+            return;
+        }
+        auto lineFormat = [](auto s1, auto s2, auto s3)
+        {
+            return fmt::format("{:4}{:^16}{:13}{:^9}{:21}{:^18}\n",
+                               " ", s1, " ", s2, " ", s3);
+        };
+
+        std::string str = "\nLIST OF ALL NON-ZERO THRESHOLD PRESSURES\n"
+                          "----------------------------------------\n"
+                          "\n";
+        str += lineFormat("FLOW FROM REGION", "TO REGION", "THRESHOLD PRESSURE");
+        str += lineFormat(std::string(16, '-'), std::string(9, '-'), std::string(18, '-'));
+        for (const auto& it : m_pressureTable) {
+            str += lineFormat(it.first.first, it.first.second, it.second.second);
```

Comment:
For defaulted threshold pressures this will print 0.0 I believe, rather than skipping them as line 299 indicates. However, that is not really the problem, as we should write the actual threshold pressures (calculated from initial state) used in the simulation for those cases, rather than just the table values from the deck, i.e. the same information that was fixed in OPM/opm-simulators#4503. So this code should probably be moved to eclgenericthresholdpressures.cc.

It has also been requested to only write the thresholds once in the symmetric case, not sure if that is taken care of here.

Reply:
symmetric is taken care of here, but will need some additional work when moved outside the class.